src/network/networkd-routing-policy-rule.c

```diff
@@ -836,6 +862,109 @@ int config_parse_routing_policy_rule_device(
         return 0;
 }
 
+int config_parse_routing_policy_rule_port_range(const char *unit,
+                                                const char *filename,
+                                                unsigned line,
+                                                const char *section,
+                                                unsigned section_line,
+                                                const char *lvalue,
+                                                int ltype,
+                                                const char *rvalue,
+                                                void *data,
+                                                void *userdata) {
+        _cleanup_(routing_policy_rule_freep) RoutingPolicyRule *n = NULL;
+        _cleanup_free_ char *word = NULL;
+        Network *network = userdata;
+        unsigned low, high;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = routing_policy_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = extract_first_word(&rvalue, &word, NULL, 0);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to extract routing policy rule port range, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        if (r == 0)
+                return 0;
+
+        r = parse_range(word, &low, &high);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse routing policy rule port range '%s'", word);
+                return 0;
+        }
+
+        if (low <= 0 || low > 65535 || high <= 0 || high > 65535) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse routing policy rule port range '%s'. Port should be greater than 0 and less than 65535.", word);
```

Comment:
Parsing range was succeed. How about the following?
"Invalid port range '%s' for routing policy rule. Port should be..."